### ISSUE TITLE
Contents gives error when image is loaded (but still works)

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#260]
I traced it to this in `Contents`:

```python
def _highlight_path(self, hl_path, tf):
    ...
    self.treeview.highlight_path(hl_path, tf, font_color=fc)
    ...
```

When I commented out the `try..except` block, I get this:

```python
2015-12-23 17:14:43,556 | E | Callback.py:88 (make_callback) | Error making callback 'image-set': 'jb510rt5q_flc[SCI,1]'
2015-12-23 17:14:43,557 | E | Callback.py:89 (make_callback) | Traceback:
  File ".../ginga-2.5.20151222205257-py2.7.egg/ginga/misc/Callback.py", line 72, in make_callback
    res = method(*cb_args, **cb_kwdargs)

  File ".../ginga-2.5.20151222205257-py2.7.egg/ginga/misc/plugins/Contents.py", line 308, in set_image_cb
    self.update_highlights(self._hl_path, new_highlight)

  File ".../ginga-2.5.20151222205257-py2.7.egg/ginga/misc/plugins/Contents.py", line 288, in update_highlights
    self._highlight_path(key, True)

  File ".../ginga-2.5.20151222205257-py2.7.egg/ginga/misc/plugins/Contents.py", line 270, in _highlight_path
    self.treeview.highlight_path(hl_path, tf, font_color=fc)

  File ".../ginga-2.5.20151222205257-py2.7.egg/ginga/qtw/Widgets.py", line 758, in highlight_path
    item = self._path_to_item(path)

  File ".../ginga-2.5.20151222205257-py2.7.egg/ginga/qtw/Widgets.py", line 750, in _path_to_item
    item = s[path[-1]].item
```

**Comment by ejeschke** [comment]
Is this from `stginga` or stock ginga reference viewer?

**Comment by ejeschke** [comment]
I've tried loading a few HST files and don't see any problem yet.

**Comment by pllim** [comment]
Pretty sure it was stock from latest master. I'll check my cfg file. Could be settings related.

**Comment by pllim** [comment]
This are the variables in `ginga.qtw.Widgets.Treeview._path_to_item()` when the traceback occurs:

```python
path =  ('Image', 'jb510rt5q_flc[SCI,1]')
self.shadow = {'Image': {'terminal': False, 'item': <PyQt4.QtGui.QTreeWidgetItem object at ...>, 'node': {}}}
self.shadow['Image'].node = {}
```

My hunch is that Contents is trying to highlight the image before it is completely loaded. The image is a 41MB multi-extension FITS.

**Comment by pllim** [comment]
I tried adding a lock here and there, but that didn't work. So I just added a check in `Treeview` (Qt). See #261.

**Comment by pllim** [comment]
Also related: I noticed that the thumbnail is not highlighted at all after initial image load, but then it works again when I cycle through the images in the channel. It turns out when the images were loaded, in `Thumbs.update_highlights()`:

```python
for thumbkey in re_hilite_set:
    # These are the variable values, so as a result, the if statement below skips the highlighting!
    # thumbkey = ('Image', 'jb510rt5q_flc[SCI,1]', '.../jb510rt5q_flc.fits')
    # self.thumbDict = {}
    if thumbkey in self.thumbDict:
        ...
```

Do you see this behavior as well, @ejeschke ? If not, I'll see after the holidays if this only affects Ginga session when I am logged into the machine remotely.

**Comment by ejeschke** [comment]
I think I missed that error message in the log without the traceback--I now see the issue.  And I think the problem is that the callback for 'image-set' happens before the callback for 'add-image'.  So there is no entry to highlight.  Working on a fix.

**Comment by ejeschke** [comment]
@pllim, can you verify that commit 8c4827d8a05779f065c540fabb1f8bc9bacdaa93 fixes this issue for you?  It should also solve the issue where the thumbnail is not highlighted when the image is initially loaded.

**Comment by pllim** [comment]
Thanks, @ejeschke ! It is fixed.

**Comment by pllim** [comment]
@ejeschke , I am seeing this error again when I open several images at once, and also auto-load extra extensions from the same images at the same time, like this:

```python
image = fv.load_image(filename, idx=extnum)
future = Future.Future()
future.freeze(fv.load_image, filename, idx=extnum)
image.set(path=filename, idx=extnum, name=cachekey, image_future=future)
fv.add_image(cachekey, image, chname=chname, silent=True)
fv.advertise_image(chname, image)
```

Do you have an idea on how to fix this? Am I doing something wrong in the codes above? I had done this before without such error, so somehow it is now incompatible with the latest changes you made?

**Comment by ejeschke** [comment]
@pllim, are you seeing this when dragging and dropping multiple images or some other open method or only with the code snippet you showed above?  Trying to reproduce now by opening multiple files, but I cannot reproduce it yet.

**Comment by ejeschke** [comment]
Could you try changing the call `make_async_gui_callback` in `Control.py:advertise_image()` to `make_gui_callback`?  Same parameters, just change the name.  Does this fix the problem?

**Comment by pllim** [comment]
Fixed (again) in #273.
